pallets/proxy-teleport/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+// This file is part of Trappist.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Disclaimer:
+// This is an experimental implementation of a pallet-xcm extension.
+// This module is not audited and should not be used in production.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+type BaseXcm<T> = pallet_xcm::Pallet<T>;
+use frame_support::{
+	dispatch::DispatchResult,
+	ensure,
+	traits::{Contains, EnsureOrigin, Get},
+	weights::Weight,
+};
+use frame_system::pallet_prelude::OriginFor;
+pub use pallet::*;
+use parity_scale_codec::Encode;
+use sp_std::{boxed::Box, vec};
+pub use xcm::{
+	latest::prelude::{
+		AssetId, ExecuteXcm, Fungibility,
+		Instruction::{
+			BurnAsset, BuyExecution, DepositAsset, DepositReserveAsset, InitiateReserveWithdraw,
+			ReceiveTeleportedAsset, Transact, WithdrawAsset,
+		},
+		Junction, Junctions, MultiAsset, MultiAssetFilter, MultiAssets, MultiLocation, OriginKind,
+		Parent, SendXcm, WeightLimit,
+		WeightLimit::Unlimited,
+		WildMultiAsset, Xcm, XcmHash,
+	},
```

Comment:
Preludes are modules where you expect all of their contents to be in scope at all times.
In general you import all the items with the `*` wildcard.
Sometimes you might import a specific item, but if you are importing so much, you can just use the wildcard.